services/ingest/tests/test_correlation_engine.py

```
@@ -145,11 +144,57 @@ async def mock_execute(stmt):
+            captured_stmts.append(stmt)
+            result = MagicMock()
+            result.scalars.return_value.all.return_value = []
+            result.scalar_one_or_none.return_value = None
+            return result
+
+        session = AsyncMock()
+        session.execute = mock_execute
+        session.flush = AsyncMock()
+        session.add = MagicMock()
+
+        def compiled_where(stmt):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `compiled_where`

(ANN202)